org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/console/JavaStackTraceHyperlink.java

```diff
@@ -183,16 +205,29 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
 				if (source == null) {
 					if (status == null) {
 						// did not find source
-						MessageDialog.openInformation(JDIDebugUIPlugin.getActiveWorkbenchShell(), ConsoleMessages.JavaStackTraceHyperlink_Information_1, NLS.bind(ConsoleMessages.JavaStackTraceHyperlink_Source_not_found_for__0__2, new String[] {typeName}));
+						MessageDialog.openInformation(JDIDebugUIPlugin.getActiveWorkbenchShell(), ConsoleMessages.JavaStackTraceHyperlink_Information_1, NLS.bind(ConsoleMessages.JavaStackTraceHyperlink_Source_not_found_for__0__2, new String[] {
+								typeName }));
 					} else {
 						JDIDebugUIPlugin.statusDialog(ConsoleMessages.JavaStackTraceHyperlink_3, status);
 					}
 				} else if (source instanceof List) { // ambiguous
+
 					@SuppressWarnings("unchecked")
 					List<Object> matches = (List<Object>) source;
 					int line = lineNumber + 1; // lineNumber starts with 0, but line with 1, see #linkActivated
-					OpenFromClipboardAction.handleMatches(matches, line, typeName, ConsoleMessages.JavaDebugStackTraceHyperlink_dialog_title);
-					return Status.OK_STATUS;
+					if (generatedLink.get() == null) { // Handles invalid links (without line number)
+						return openClipboard(matches, line, typeName);
+					}
+					try {
+						return processAmbiguousResults(matches, typeName, lineNumber);
+					} catch(Exception e) {
+						String link = generatedLink.get();
```

Comment:
it would be better to avoid multiple generatedLink.get(). Take an initial local copy and work with that, as the value could theoretically change over time.

Reply:
done.